addons/mail/static/src/web/messaging_menu/messaging_menu.js

```diff
@@ -113,47 +111,35 @@ export class MessagingMenu extends Component {
         if (tab !== "all") {
             threads = threads.filter(({ type }) => this.tabToThreadType(tab).includes(type));
         }
-        threads.sort((a, b) => {
-            if (a.correspondent === this.store.odoobot) {
-                return 1;
-            }
-            if (!a.newestPersistentMessage?.datetime) {
-                return -1;
-            }
-            if (!b.newestPersistentMessage?.datetime) {
-                return 1;
-            }
-            return b.newestPersistentMessage.datetime - a.newestPersistentMessage.datetime;
-        });
-        const previews = [];
-        for (const thread of threads) {
-            const { newestMessage, newestNeedactionMessage } = thread;
-            if (thread.is_pinned) {
-                const message = newestMessage;
-                previews.push({
-                    id: `preview-${thread.localId}`,
-                    count: thread.message_unread_counter,
-                    imgUrl: thread.imgUrl,
-                    hasMarkAsReadButton: thread.message_unread_counter > 0,
-                    message,
-                    thread,
-                    isNeedaction: false,
-                });
-            }
-            if (newestNeedactionMessage) {
-                const message = newestNeedactionMessage;
-                previews.push({
-                    id: `preview-needaction-${thread.localId}`,
-                    count: thread.needactionMessages.length,
-                    imgUrl: message.module_icon,
-                    hasMarkAsReadButton: true,
-                    message,
-                    thread,
-                    isNeedaction: true,
-                });
-            }
-        }
-        return previews;
+        return threads
+            .sort((a, b) => {
```

Comment:
It could be useful to document the way this need to be sorted, the code isn't easy to read nor self explanatory. Something like a hierarchy that display the order of the messaging menu. Will be easier to debug/modify in the future with a clear reference.